evm/p2p/les.py

```diff
@@ -6,14 +18,159 @@
 
 class Status(Command):
     _cmd_id = 0
+    decode_strict = False
+    # A list of (key, value) pairs is all a Status msg contains, but since the values can be of
+    # any type, we need to use the raw sedes here and do the actual deserialization in
+    # decode_payload().
+    structure = sedes.CountableList(sedes.List([sedes.binary, sedes.raw]))
+    # The sedes used for each key in the list above.
+    items_sedes = {
+        'protocolVersion': sedes.big_endian_int,
+        'networkId': sedes.big_endian_int,
+        'headTd': sedes.big_endian_int,
+        'headHash': sedes.binary,
+        'headNum': sedes.big_endian_int,
+        'genesisHash': sedes.binary,
+        'serveHeaders': None,
+        'serveChainSince': sedes.big_endian_int,
+        'serveStateSince': sedes.big_endian_int,
+        'txRelay': None,
+        'flowControl/BL': sedes.big_endian_int,
+        'flowControl/MRC': sedes.CountableList(
+            sedes.List([sedes.big_endian_int, sedes.big_endian_int, sedes.big_endian_int])),
+        'flowControl/MRR': sedes.big_endian_int,
+    }
+
+    def decode_payload(self, rlp_data):
+        data = super(Status, self).decode_payload(rlp_data)
+        # The LES/Status msg contains an arbitrary list of (key, value) pairs, where values can
+        # have different types and unknown keys should be ignored for forward compatibility
+        # reasons, so here we need an extra pass to deserialize each of the key/value pairs we
+        # know about.
+        decoded = {}
+        for key, value in data:
+            # The sedes.binary we use in .structure above will give us a bytes value here, but
+            # using bytes as dictionary keys makes it impossible to use the dict() constructor
+            # with keyword arguments, so we convert them to strings here.
+            key = key.decode('ascii')
+            if key not in self.items_sedes:
+                continue
+            item_sedes = self.items_sedes[key]
+            if item_sedes is not None:
+                decoded[key] = item_sedes.deserialize(value)
+            else:
+                decoded[key] = value
+        return decoded
+
+    def encode_payload(self, data):
+        response = []
+        for key, value in data.items():
```

Comment:
What do you think about sorting this by keys so that our return value is always consistently ordered?